### PR TITLE
completion: add additional documentation

### PR DESCRIPTION
**Pull request by oliver-sanders**
**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
(rebased)

File: src/reference/changes.rst [hunk 1]
```diff
@@ -132,6 +132,41 @@ The graph view now has an option to group tasks by cycle point.
 .. image:: changes/cylc-graph-group-by-cycle-point.png
    :width: 100%
 
+
+Completion Expressions
+^^^^^^^^^^^^^^^^^^^^^^
+
+When a task finishes, its outputs are validated against a "completion
```
Comment (hjoliver) [inline src/reference/changes.rst:139]: Yikes!  

```suggestion
When a task achieves a final status, its outputs are validated against a "completion
```

File: src/reference/changes.rst [hunk 2]
```diff
@@ -132,6 +132,41 @@ The graph view now has an option to group tasks by cycle point.
...
+:term:`required outputs <required output>`.
+If a task fails this validation check it is said to have "incomplete outputs"
+and will be retained in the :term:`active window` pending user intervention.
+
+This completion expression is generated automatically from the graph.
+By default, tasks are expected to succeed, if you register any additional
+required output in the graph, then these must also
+be produced.
+
+At Cylc 8.3.0 it is now possible to manually configure this completion
```
Comment (hjoliver) [inline src/reference/changes.rst:150]: No biggie, but we'd normally not say this here (put an "added in version ..." note instead?)

File: src/reference/changes.rst [hunk 3]
```diff
@@ -132,6 +132,41 @@ The graph view now has an option to group tasks by cycle point.
...
+expression for finer control. This is particularly useful for anyone using
+:term:`custom outputs <custom output>`.
+
+For example, ``mytask`` must produce one of the outputs ``x`` or ``y`` to pass
+the configured completion expression configured here:
```
Comment (hjoliver) [inline src/reference/changes.rst:155]:
```suggestion
the completion expression configured here:
```
(too many "configured")

File: src/user-guide/writing-workflows/scheduling.rst [hunk 4]
```diff
@@ -1894,12 +1894,12 @@ A more realistic example might have several tasks on each branch. The
...
-Message Trigger Example
-^^^^^^^^^^^^^^^^^^^^^^^
+Custom Outputs
+^^^^^^^^^^^^^^
 
-Branching is particularly powerful when using :ref:`MessageTriggers` (i.e.
-:term:`custom outputs <custom output>`) to define multiple parallel paths in
-the graph.
+Branching is particularly powerful when using
+:term:`custom outputs <custom output>` to define multiple parallel paths in the
```
Comment (hjoliver) [inline src/user-guide/writing-workflows/scheduling.rst:1901]:
```suggestion
:term:`custom outputs <custom output>` to define alternate parallel paths in the
```

(not sure, but we probably want to distinguish this from "normal" multiple parallel paths in the graph?)

File: src/user-guide/writing-workflows/scheduling.rst [hunk 5]
```diff
@@ -1972,36 +1972,62 @@ three custom outputs:
...
+""""""""""""""""""""""
 
-When using message triggers in this way there are two things to be aware of:
+.. cylc-scope:: flow.cylc[runtime][<namespace>]
 
-1. Message triggers are not exit states.
+The :cylc:conf:`completion` configuration above is optional, it adds a basic
+validation check which ensures that at least one of the three custom outputs is
+produced when the task runs. This protects you against the possibility that
+none of the outputs are produced e.g. due to a task design error.
```
Comment (hjoliver) [inline src/user-guide/writing-workflows/scheduling.rst:1993]:
```suggestion
none of the outputs are produced e.g. due to a task implementation error.
```
?

**Review by hjoliver — APPROVED** [review]
Couple of minor suggestions

File: src/reference/changes.rst (hunk 2 above)
Comment (wxtim) [reply src/reference/changes.rst:150]:
```suggestion
.. versionadded:: 8.3.0

   It is now possible to manually configure this completion
```

Comment (wxtim) [reply src/reference/changes.rst:150]: I'm not sure we'd do that in `changes.rst` - or the whole document would be a storm of `.. versionadded::` directives.

**Pull request merged**